middle_end/flambda2/types/grammar/type_grammar.ml

```diff
@@ -3093,24 +3144,30 @@ end
 module Head_of_kind_naked_immediate = struct
   type t = head_of_kind_naked_immediate
 
+  let create ~immediates ~relations = { immediates; relations }
+
   let create_naked_immediate imm =
-    Naked_immediates (Targetint_31_63.Set.singleton imm)
+    { immediates = Known (Targetint_31_63.Set.singleton imm);
+      relations = RelationSet.empty
+    }
 
   let create_naked_immediates imms : _ Or_bottom.t =
     if Targetint_31_63.Set.is_empty imms
     then Bottom
-    else Ok (Naked_immediates imms)
+    else Ok { immediates = Known imms; relations = RelationSet.empty }
 
   let create_naked_immediates_non_empty imms =
     if Targetint_31_63.Set.is_empty imms
     then
       Misc.fatal_error
         "Head_of_kind_naked_immediates.create_naked_immediates_non_empty";
-    Naked_immediates imms
+    { immediates = Known imms; relations = RelationSet.empty }
 
-  let create_is_int ty = Is_int ty
+  let create_is_int name =
+    { immediates = Unknown; relations = RelationSet.singleton (Is_int name) }
```

Comment:
Shouldn't immediates be the set {#0, #1} here?

Reply:
Yes, that would work, although I'm not sure if that is useful. If you look at the provers for instance then we do more clever things it this is still unknown.